src/main/java/com/igot/cb/discussion/service/impl/DiscussionServiceImpl.java

```diff
@@ -477,4 +495,59 @@ public List<Object> fetchDataForKeys(List<String> keys) {
                 })
                 .collect(Collectors.toList());
     }
+
+    public List<Object> fetchUserFromprimary(List<String> userIds) {
+        List<Object> userList = new ArrayList<>();
+        Map<String, Object> propertyMap = new HashMap<>();
+        propertyMap.put(Constants.ID, userIds);
+        List<Map<String, Object>> userInfoList = cassandraOperation.getRecordsByPropertiesWithoutFiltering(
+                Constants.KEYSPACE_SUNBIRD, Constants.USER_TABLE, propertyMap,
+                Arrays.asList(Constants.PROFILE_DETAILS, Constants.FIRST_NAME, Constants.ID), null);
```

Comment:
what would be the number of ids sent in the userIds -- will cassandra be able to handle the load??
do we need to read it in small amounts?

Reply:
there will be very minimum number of ids passed as of now redis data is present for all the users if only redis data is not present we are fetching it from cassandra.
currently if the user is created then the data is not present in the redis 
for the newly created users we need to fetch from cassandra, it will updated in redis next day,